TST: unit tests for plot utilities

aburrell opened the pull request:
Unit tests and upgrades to the plot utilities.

aburrell committed:
ENH: improved slicing robustness

Improved the robustness of `get_cut_index` by:
- adding a boolean kwarg that specifies whether or not the provided cut value is an index,
- raising ValueErrors when input is out of bounds,
- warning users about suspect input, but allow the routine to work, and
- improving the docstrings.

aburrell committed:
TST: added slicing unit tests

Added unit tests for the `get_cut_index` data preparation function.

aburrell committed:
DOC: updated TEC docstring

Updated the TEC docstring to conform to standard formatting better, define all acronyms, and have a section for errors.

aburrell committed:
BUG: fixed TEC bugs

Fixed a bug in the TEC calculation, where `shape` was not called correctly.  Extended `ialt_min` input to allow negative indices.

aburrell committed:
TST: added TEC unit tests

Added unit tests for the `calc_tec` function.

aburrell committed:
ENH: improved movie functions

Improved the compatibility and extended the user options for the movie creation functions.

aburrell committed:
TST: created movie tests

Created a unit test class for the movie functions.

aburrell committed:
DEP: fixed deprecation warning

Fixed a deprecation warning, which now requires all strings with a backslash are raw.

aburrell committed:
TST: added movie unit tests

Finished adding movie unit tests.

aburrell committed:
TST: add ffmpeg to CI

Added the installation of ffmpeg to GitHub Actions using https://github.com/marketplace/actions/setup-ffmpeg as a guide.

aburrell committed:
Merge branch 'develop' into plot_tests

qalshidi commented:
Are the PEP8 violations to be fixed in a different PR?

aburrell commented:
No they need to be fixed before merging.  That's why I didn't ping anyone for a review yet.

aburrell committed:
STY: fixed flake8 errors

Fixed several flake8 errors and added a test for the expected output image directory.

aburrell committed:
TST: added rename loop

Added a loop for renaming the temporary file, a potential solution for the Windows error.

aburrell committed:
TST: added neg lon test

Added a unit test for negative longitudes in the `ut_to_lt` method.

aburrell committed:
TST: fixed GA bug

Fixed a bug in GitHub Actions designed to catch spammers on the MacOS.  Following: https://github.com/actions/virtual-environments/issues/602.

aburrell committed:
TST: added more windows fixes

Added more windows fixes to the tests.

aburrell committed:
BUG: expected `token`

`setup-ffmpeg` expects the `token` key instead of the `repo-token` key.

aburrell committed:
TST: replaced all renames

Replaced all calls to `os.rename` calls with a local function that allows multiple retries when renaming for Windows operating systems.

aburrell committed:
DOC: expanded docstrings

Added note in docstrings on expected behaviour for windows systems.

aburrell committed:
TST: skip windows failures

Skip tests that are not expected to succeed on Windows systems.

aburrell committed:
BUG: updates skipif statements

Updated the skipif statements to include the expected values based on the documentation.

aburrell committed:
TST: skip failing windows test

Skipped the failing Windows test, unable to solve permission errors.

ponderbrandon1 commented on aetherpy/plot/data_prep.py:
```diff
@@ -44,72 +47,68 @@ def get_cut_index(lons, lats, alts, cut_val, cut_coord='alt'):
+        icut = cut_val
+    else:
+        if cut_val < z_coord.min() or cut_val > z_coord.max():
+            raise ValueError('Requested cut is outside the coordiante range')
```
```suggestion
            raise ValueError('Requested cut is outside the coordinate range')
```

ponderbrandon1 commented on aetherpy/tests/test_plot_movie.py:
```diff
@@ -0,0 +1,241 @@
+        image_files = os.path.join(self.movie_dir, "".join([
+            self.filebase, "%04d", self.fileext]))
+
+        # Create the move file
```
```suggestion
        # Create the movie file
```

ponderbrandon1 commented on aetherpy/tests/test_plot_prep.py:
```diff
@@ -0,0 +1,226 @@
+        """Initialize clean test environment."""
+
+        self.in_coords = {"lon": np.arange(-180, 180, 1.0),
+                          "lat": np.arange(-89, 90, 0.5),
```
Why shouldn't this go from -90 to 90?

aburrell replied:
This goes from -89 to 89.5 because the model doesn't go up to +/-90, it stops beforehand.  I believe that longitude can't be defined at +/- 90 degrees and that's why it isn't included.

ponderbrandon1 commented on aetherpy/tests/test_utils_time.py:
```diff
@@ -98,20 +98,29 @@ def update_lon(self, multi_lon):
+    @pytest.mark.parametrize("lon_sign", [1, -1])
     @pytest.mark.parametrize("multi_lon", [True, False])
-    def test_ut_to_lt(self, multi_lon):
+    def test_ut_to_lt(self, lon_sign, multi_lon):
```
Does this work for non-Earth planets? Should we add in tests for this?

aburrell replied:
Good question!  I do not think the routine would work for other planets because it uses datetime input, and so is converting from universal time to solar local time (using two earth-based assumptions: location of prime meridian and hours/degrees longitude).  Introducing other planets would be outside the scope of this pull request, but be a good thing to do.  Would you like to create an issue in the repository for this?

aburrell replied:
@ponderbrandon1 you need to press the resolve button here for the merge to go forward.

aburrell commented on aetherpy/tests/test_plot_prep.py:
```diff
@@ -0,0 +1,226 @@
+        """Initialize clean test environment."""
+
+        self.in_coords = {"lon": np.arange(-180, 180, 1.0),
+                          "lat": np.arange(-89, 90, 0.5),
```
```suggestion
                          "lat": np.arange(-89.5, 90, 0.5),
```

Even out the latitude range for symmetry.

aburrell committed:
TST: fix test long range

Fixed the test longitude range to be symmetric and fixed spelling errors.

Co-authored-by: Brandon Ponder <39278884+ponderbrandon1@users.noreply.github.com>

aaronjridley commented:
Not sure if reply in e-mail works.... this is a test, I guess.
I always define longitude from 0 - 360 and not from -180 - 180.
Also, in GITM and maybe other models, in order to plot a complete circle, you need overlap in longitude, so often the longitudes will extend past 360 or below 0. For example, in GITM, if there is 4 deg longitude resolution, then the ghost cells are at -6°, -2° and 362° and 366°.
Aaron
Aaron Ridley
Professor
734-764-5727 (tel:734-764-5727)
1416 Space Research Building, Ann Arbor (https://maps.google.com/?q=1416%20Space%20Research%20Building%2C%20Ann%20Arbor)

On Feb 15 2022, at 8:38 am, Angeline Burrell ***@***.***> wrote:
>
>
> @aburrell commented on this pull request.
> In aetherpy/tests/test_plot_prep.py (https://github.com/AetherModel/aetherpy/pull/8#discussion_r806838944):
>
> > +
>
> +import logging
>
> +import numpy as np
>
> +import pytest
>
> +
>
> +from aetherpy.plot import data_prep
>
> +
>
> +
>
> +class TestDataPrep(object):
>
> + """Unit tests for data preparation functions."""
>
> +
>
> + def setup(self):
>
> + """Initialize clean test environment."""
>
> +
>
> + self.in_coords = {"lon": np.arange(-180, 180, 1.0),
>
> + "lat": np.arange(-89, 90, 0.5),
>
> ⬇️ Suggested change
> - "lat": np.arange(-89, 90, 0.5),
>
> + "lat": np.arange(-89.5, 90, 0.5),
>
> Even out the latitude range for symmetry.
> —
> Reply to this email directly, view it on GitHub (https://github.com/AetherModel/aetherpy/pull/8#pullrequestreview-882994993), or unsubscribe (https://github.com/notifications/unsubscribe-auth/AAOYHCV2WFMKF5S5LMSGJD3U3JJNHANCNFSM5NRZ3F3Q).
> Triage notifications on the go with GitHub Mobile for iOS (https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675) or Android (https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub).
> You are receiving this because you are subscribed to this thread.
>

aburrell commented on aetherpy/tests/test_plot_prep.py:
```diff
@@ -0,0 +1,226 @@
+    def setup(self):
+        """Initialize clean test environment."""
+
+        self.in_coords = {"lon": np.arange(-180, 180, 1.0),
```
```suggestion
        # Set the testing latitude and longitude range to
        # include all locations where model can uniquely
        # define both latitude and longitude
        self.in_coords = {"lon": np.arange(-180, 180, 1.0),
```

aburrell committed:
DOC: updated test comments

Updated the test set up to explain why the limits were set the the chosen values.

aburrell committed:
BUG: fixed spelling in test

Fixed spelling error corrected in file but not text.  Also increased tested longitude range.

aburrell commented:
@aaronjridley the email did work!  The routine can take either longitude range, so I extended it to test for both.
